### ISSUE TITLE
Where should the shapes of chat apps be defined?

### ISSUE THREAD

**Issue by labra** [labra/solid-shapes#1]
In order to obtain some minimal interoperability between solid chat apps, it would be interesting to have some shapes defined for the chat messages. 
In [this folder](https://github.com/labra/solid-shapes/tree/master/examples), I reverse-engineered the shapes of the [solid chat pane](https://github.com/solid/solid-panes/tree/master/chat).
However, I think there should be a better place to store those shapes and it would be better if those properties had some consensus within the Solid community.

**Comment by timbl** [comment]
Yes it isn't obvious where to store shapes - they are a new thing and both part of the standard and part of the code .. but for the mean time we just put some in
https://github.com/solid/chat-pane/blob/master/shapes/chat-shapes.ttl
I'm sorry no one pointed you at them

**Comment by timbl** [comment]
We have shapes for chat, and for vcard contact data, but should be making shapes for everything.  We have solid-ui forms for various things, which are closely related to shapes.

**Comment by RubenVerborgh** [comment]
Indeed, and we need work on how to explicitize those shapes, share them among apps, etc. We need to think about them like ontologies.

**Comment by labra** [comment]
> Yes it isn't obvious where to store shapes - they are a new thing and both part of the standard and part of the code .. but for the mean time we just put some in
> https://github.com/solid/chat-pane/blob/master/shapes/chat-shapes.ttl
> I'm sorry no one pointed you at them

Indeed, I knew about the existence of those shapes but thought they were a draft. We had a small conversation about those shapes [in gitter on 18th Feb](https://gitter.im/solid/app-development?at=5c69f12d604f233ab629527e). 

One problem of those shapes is that they are not valid SHACL shapes. That's why I had created this repository which contains valid SHACL shapes and translated also those shapes to ShEx as it is more readable.

I have just done a [Pull Request](https://github.com/solid/chat-pane/pull/15) to the `chat-pane` repository including those shapes.

Anyway, it would be interesting to establish some kind of procedure to discuss about those shapes...

For example, why are some properties mandatory while others are optional, and which shapes can be reused or not. It seems to me that the chat shapes contain some references to user preferences which could probably be refactored to other place. What is the right place to have those discussions?

I agree that we should think about the shapes as first class citizens and have some place where we could share them and make them more explicit.

**Comment by Julian-Cole** [comment]
If mySolidApp/  is assumed to be the location of the index.html of my app, why couldn't mySolidApp/shapes.ttl   be the home for the definition of the shapes used by my app?

Just a thought on the structure of the code for the solid-panes;  if you completely separated the chat code into a single api which was called from the front-end,  it would make it incredibly easy for others  to make basic shape-data compatible chat apps etc;    In the same way that solid-file-client is a nice standard way to manipulate pod data.  As the solid-panes become more sophisticated, so does the api set.

By defining a large varied standard api set over time you would hopefully end up with all the basic building blocks needed to build all the required core of web apps.
